aws-datastore/src/test/java/com/amplifyframework/datastore/storage/InMemoryStorageAdapter.java

```diff
@@ -187,6 +191,43 @@ public void query(
         onSuccess.accept(deletion);
     }
 
+    @SuppressWarnings("unchecked") // item.getClass() -> Class<?>, but type is T. So cast as Class<T> is OK.
+    @Override
+    public <T extends Model> void delete(
```

Comment:
Any opportunity to consolidate the implementation with the storage adapter's other delete method(s)?

Reply:
this one behaves completely differently than the other delete method unfortuantely. If anything, the all the other methods where a single data is being written/removed (i.e. `save(Model)` and `delete(Model)`)  are similar enough to each other that it's plausible to consolidate those. I made an effort to do something along that line actually in my last refactoring PR, but didn't remove any methods.